### ISSUE TITLE
An error occurred while handling textDocument/codeAction request: NoSuchMethodError: The getter 'id' was called on null.

### ISSUE THREAD

**Issue by ascott18** [dart-lang/sdk#47089]
I'm getting the following error whenever I hover over some underlined problems in my code in VS Code. 

Many problem types don't cause this, but it is consistently reproducible on, for example, this problem:
![image](https://user-images.githubusercontent.com/5017521/131903076-ac98f474-7266-44c6-8a73-4af3de1d6750.png)

```
[Error - 12:13:03 PM] Request textDocument/codeAction failed.
  Message: An error occurred while handling textDocument/codeAction request
  Code: -32001 
[Error - 12:13:03 PM] An error occurred while handling textDocument/codeAction request: NoSuchMethodError: The getter 'id' was called on null.
Receiver: null
Tried calling: id
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:54:5)
#1      FixProcessor._addFixFromBuilder (package:analysis_server/src/services/correction/fix_internal.dart:1741:22)
#2      FixProcessor._addFromProducers.compute (package:analysis_server/src/services/correction/fix_internal.dart:1768:9)
<asynchronous suspension>
#3      FixProcessor._addFromProducers (package:analysis_server/src/services/correction/fix_internal.dart:1782:11)
<asynchronous suspension>
#4      FixProcessor.compute (package:analysis_server/src/services/correction/fix_internal.dart:1724:5)
<asynchronous suspension>
#5      DartFixContributor.computeFixes (package:analysis_server/src/services/correction/fix_internal.dart:185:19)
<asynchronous suspension>
#6      CodeActionHandler._getFixActions (package:analysis_server/src/lsp/handlers/handler_code_actions.dart:311:23)
<asynchronous suspension>
#7      Future.wait.<anonymous closure> (dart:async/future.dart)
<asynchronous suspension>
#8      CodeActionHandler._getCodeActions (package:analysis_server/src/lsp/handlers/handler_code_actions.dart:261:21)
<asynchronous suspension>
#9      ServerStateMessageHandler.handleMessage (package:analysis_server/src/lsp/handlers/handlers.dart:187:22)
<asynchronous suspension>
#10     LspAnalysisServer.handleMessage.<anonymous closure> (package:analysis_server/src/lsp/lsp_analysis_server.dart:302:26)
<asynchronous suspension>

```

* Dart SDK Version (`dart --version`)
```
PS> dart --version
Dart SDK version: 2.13.4 (stable) (Wed Jun 23 13:08:41 2021 +0200) on "windows_x64"
```

**Comment by bwilkerson** [comment]
Thanks for the report.

It's a little hard to tell because the stack trace doesn't line up with the current version of the code, but based on the method in which the exception occurred and the name of the getter being invoked it looks like this has been fixed:

```dart
  void _addFixFromBuilder(ChangeBuilder builder, CorrectionProducer producer) {
    var change = builder.sourceChange;
    if (change.edits.isEmpty) {
      return;
    }

    var kind = producer.fixKind;
    if (kind == null) {
      return;
    }

    change.id = kind.id; // <--
    change.message = formatList(kind.message, producer.fixArguments);
    fixes.add(Fix(kind, change));
  }
```

There doesn't appear to be any way for `id` to be invoked on `null`.

However, I don't know how to verify that the null check, which was added on 4/12/21, isn't in 2.13.4. The dates make that seem unlikely, but I can't think of any other way you could be seeing what you're seeing.

**Comment by ascott18** [comment]
I deleted flutter/bin/cache and then had the Dart SDK reinstalled. Still same issue.

`flutter/bin/cache/dart-sdk/bin/snapshots/analysis_server.dart.snapshot` has a file date of 6/30/2021 (as do all the other files in that folder).

Flutter + Dart versions from doctor -v output:

```
[√] Flutter (Channel stable, 2.2.3, on Microsoft Windows [Version 10.0.19043.1165], locale en-US)
    • Flutter version 2.2.3 at C:\flutter
    • Framework revision f4abaa0735 (9 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4
```

**Comment by bwilkerson** [comment]
Do you have reproduction steps you could send? Even an update stack trace would likely be helpful.

**Comment by ascott18** [comment]
In looking at the tag list on the commit that introduced that check (https://github.com/dart-lang/sdk/commit/050fa8a70a5c4b4df36d70176a00eb3bc3ae5369), `2.13.4` is not one of them. In fact, that commit has never been included in any non `-dev` tag.

**Comment by bwilkerson** [comment]
Yeah, I don't think those tags are accurate. We aren't developing directly on GitHub anymore, the source is somewhere else and mirrored to GitHub. My guess is that the non-dev tags aren't being copied over for some reason.

**Comment by DanTup** [comment]
This appears fixed in the changeset linked above. The changeset was not included in v2.13.x SDKs:

https://github.com/dart-lang/sdk/commits/2.13.4

But was included in v2.14.x SDKs:

https://github.com/dart-lang/sdk/commits/2.14.0?after=4c8a4f0d7ad055fa7dea5e80862cd2074f4454d3+2290&branch=2.14.0&qualified_name=refs%2Ftags%2F2.14.0

I verified the current latest code also still has the null check noted above.

Please post back if you see it on an SDK >= v2.14.0.
